src/Network/Minio.hs

```diff
@@ -273,9 +273,10 @@ getObject :: Bucket -> Object -> GetObjectOptions
 getObject bucket object opts = snd <$> getObject' bucket object []
                                (gooToHeaders opts)
 
--- | Get an object's metadata from the object store.
-statObject :: Bucket -> Object -> Minio ObjectInfo
-statObject = headObject
+-- | Get an object's metadata from the object store. It accepts the
+-- same options as GetObject.
+statObject :: Bucket -> Object -> GetObjectOptions -> Minio ObjectInfo
+statObject b o opts = headObject b o $ gooToHeaders opts
```

Comment:
If `statObject` needs only `[HT.Header]` as input why do we supply `GetObjectOptions` which has other fields that we don't use here. We could have a newtype for `StatObjectOptions` which has a `[HT.Header]`

Reply:
We do use it - the settings object gets converted to headers